core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3104,6 +3104,9 @@ object ZIO extends ZIOCompanionPlatformSpecific {
   def fromOption[A](v: => Option[A]): IO[Option[Nothing], A] =
     effectTotal(v).flatMap(_.fold[IO[Option[Nothing], A]](fail(None))(succeedNow))
 
+  def fromOptionOrFailUnit[A](v: => Option[A]): IO[Unit, A] =
```

Comment:
Should we call `getOrFailUnit` for symmetry with existing `getOrFail` constructor?